src/ConfigSpace/functional.py

```diff
@@ -0,0 +1,414 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+from typing import TYPE_CHECKING, TypeVar
+from typing_extensions import overload
+
+import numpy as np
+from more_itertools import roundrobin
+
+from ConfigSpace.types import Number, f64, i64
+
+if TYPE_CHECKING:
+    from ConfigSpace.types import Array, Mask
+
+
+def center_range(
+    center: int,
```

Comment:
What happens if the difference from the lower to the center is smaller than from the center to higher? Is this an issue?

Reply:
Nope, `roundrobin()` will start just giving those from the `above_center` only after it can no longer yield equally between `below_center` and `above_center`